src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy

```diff
@@ -0,0 +1,57 @@
+package com.avioconsulting.mule.linter.model
+
+import groovy.xml.slurpersupport.GPathResult
+
+/**
+ * This class represents a Mule Configuration XML file.
+ */
+class ConfigurationFile extends ProjectFile {
+
+    MuleXmlParser parser
+    private final GPathResult configXml
+    private final Boolean exists
+
+    ConfigurationFile(File file) {
+        super(file)
+        if (file.exists()) {
+            exists = true
+            parser = new MuleXmlParser()
+            configXml = parser.parse(file)
+        } else {
+            exists = false
+        }
+    }
+
+    List<MuleComponent> findComponents(String componentType, String namespace) {
+        List<MuleComponent> componentList = []
+        GPathResult[] comps = configXml.depthFirst().findAll {
+            it.name() == componentType && it.namespaceURI() == namespace
+        }
+        comps.each { comp ->
+            Map<String, String> atts = [:]
+            comp[0].attributes.each {
+                atts.put(it.key, it.value)
+            }
+            componentList.add(new MuleComponent(atts))
+            //TODO this doesn't account for child components...
+        }
+    }
+
+    List<LoggerComponent> findLoggerComponents() {
+        List<LoggerComponent> loggerComponents = []
+        GPathResult[] loggers = configXml.depthFirst().findAll {
+            it.name() == 'logger' && it.namespaceURI() == 'http://www.mulesoft.org/schema/mule/core'
```

Comment:
Shouldn't this use findComponents?  We don't want to repeat logic for each type of component if the only difference is the name/namespace in the logic

Reply:
Made more generic.... componentName and componentNamespace are coming in Nisarg's branch, didn't want to make too many clashing changes here...